Clown Magboots

datlo opened the pull request:
Adds Clown Magboots as a job specific traitor clown item.

"A pair of modified clown shoes fitted with an advanced magnetic traction system. Look and sound exactly like regular clown shoes unless closely inspected."

Costs 3 TC, enabling the gravity pull does NOT slow you down, although it isn't any faster than regular magboots since you're slowed due to them being clown shoes anyway.

There is no visible signs of the shoes not being regular clown shoes other than the clown ignoring atmos and honking along as he walks all over banana peels.

Examining the shoes reveal that they are magnetic.

TODO 

- check if I didn't bork the changelog thing
- Add the shoes to contraband items in Space Law if merged (Wiki only?)

WHY 

Having to choose between maximum honks and optimising your ability to traitor bothers the Honkmother.

Costs the same amount as Blood red magboots, but I feel it is fair since it is a job specific traitor item.

:cl:
rscadd: Syndicate clowns can now purchase Clown Magboots. Keep honking through slips and atmos!
/:cl:

Kyep commented on code/modules/clothing/shoes/magboots.dm:
```diff
@@ -53,4 +53,25 @@
+	item_state = "clown_shoes"
+	slowdown = SHOES_SLOWDOWN+1
+	item_color = "clown"
+	var/footstep = 1	//used for squeeks whilst walking
```
Does anything use this variable? If the actual clown shoes do, then their code does not automatically apply to this.

datlo replied:
The comment was from the clown shoes. I believed that the var was called from some other file checking to make sounds when the player moves, but commenting it out does not seem to have any effects, on either my shoes or the original clown shoes. The line could probably be removed from the original shoes aswell.

Kyep commented on code/datums/uplink_item.dm:
```diff
@@ -129,6 +129,14 @@ var/list/uplink_items = list()
 
+/datum/uplink_item/jobspecific/clownmagboots
+	name = "Clown Magboots"
+	desc = "A pair of modified clown shoes fitted with an advanced magnetic pull system. Look and sound exactly like regular clown shoes unless closely inspected."
```
Perhaps 'traction' rather than 'pull'.

datlo replied:
All the other descriptions in magboots.dm mention the boots' "magnetic pull", so I went with that, although thinking about it traction does fit better in this sentence.

Kyep commented on code/modules/clothing/shoes/magboots.dm:
```diff
@@ -53,4 +53,25 @@
+	shoe_sound = "clownstep"
+	origin_tech = "magnets=4;syndicate=2"
+
+/obj/item/clothing/shoes/magboots/clown/attack_self(mob/user) //redefine toggle to remove slowdown and icon updating
```
Is this necessary, given that they are a child object of the shoes/magboots?
Also, what happens if you toggle their status? Does it change their icon, like normal magboots do? Is it meant to?

datlo replied:
The issue is that the original magboots toggle reset the player's speed back to default instead of back to whatever the previous speed was. I could fix this by adding a slowdown_passive var and modify the toggle accordingly, but this would be modifying the code of the original magboots, which I wasn't 100% comfortable with given this is my first PR. This felt like a better solution, although it does duplicate a little code.

When you toggle the shoes, the icon is not changed. This is intended behaviour, so that the shoes are not easily spotted if removed unless they are actively inspected. The player does get the message "You activate/disable the mag-pulse traction system", so although they'll have to be more careful, they know whether it's disabled or enabled. You can also examine the shoes while equipped, and it'll tell you the status of the mag-pulse.

Citinited replied:
I've not played with magboot code myself but couldn't you just do `slowdown_active = SHOES_SLOWDOWN` and not have to update the `attack_self` proc? (Similar to how advance magboots does it in https://github.com/datlo/Paradise/blob/8a909e4da5a5a443dd45af4ec9f64592e4537108/code/modules/clothing/shoes/magboots.dm#L38)

Or even just make these clown magboots a subtype of `obj/item/clothing/shoes/magboots/advance`

datlo replied:
Unless I'm misunderstanding you, this doesn't work, because the original magboots proc returns the speed to default speed when deactivated. We want the proc to return to SHOES_SLOWDOWN+1, or, as I did it, cut out slowing up and down on activation entirely.

Tayyyyyyy replied:
To reduce code duplication, could you maybe call the parent proc and then return it to the correct slowdown after?

Citinited replied:
In that case, do something like the following:

```
/obj/item/clothing/shoes/magboots/clown/attack_self(mob/user)
    ..()
    if(!magpulse)
        slowdown = initial(slowdown)
```
This prevents duplicating code whilst also resetting the slowdown

datlo commented:
I've taken your comments into account and rewrote the code to avoid code duplication and make it more modular in the future should anyone add even more magboots into the game. 

I've also added sprites! Little red light gets turned on when the shoes are activated, on both the icon and the mob sprite.

Unless there's a bug I haven't caught or something I haven't thought of, the Clown Magboots are ready to be added into the server if you guys want them.

Crazylemon64 commented:
LGTM
